axum/src/middleware/from_fn.rs

```diff
@@ -23,6 +23,8 @@ use tower_service::Service;
 /// 3. Take [`Next`](Next) as the final argument.
 /// 4. Return something that implements [`IntoResponse`].
 ///
+/// The last extractor (i.e. the second last argument) must be [`Request`] or some type implementing [`FromRequest`].
+///
```

Comment:
Why not put it inside the list between items 2 and 3? Seems like a better place since it already talks specifically about the arguments.

It might be worth changing the current 2. to read "take zero or more `FromRequestParts` extractors" and then "take exactly one `FromRequest` extractor".

Reply:
Sounds like a good suggestion. Fixed.